src/fdc3/dotnet/DesktopAgent/src/DesktopAgent/DependencyInjection/Fdc3DesktopAgentBuilder.cs

```diff
@@ -0,0 +1,45 @@
+﻿/*
+ * Morgan Stanley makes this available to you under the Apache License,
+ * Version 2.0 (the "License"). You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership. Unless required by applicable law or agreed
+ * to in writing, software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+
+using Microsoft.Extensions.DependencyInjection;
+
+namespace MorganStanley.ComposeUI.Fdc3.DesktopAgent.DependencyInjection;
+
+public class Fdc3DesktopAgentBuilder
+{
+    public IServiceCollection ServiceCollection { get; }
+
+    public Fdc3DesktopAgentBuilder(IServiceCollection serviceCollection)
+    {
+        ServiceCollection = serviceCollection;
+    }
+
+    public Fdc3DesktopAgentBuilder WithUserChannel(string channelId)
```

Comment:
Now that the user channel can be set in options, this method can be removed. `Fdc3DesktopAgent.StartAsync` should directly call `AddUserChannel` if the `ChannelId` is set in options (in other words, setting the channel shouldn't be a builder action any more).